packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -894,6 +897,10 @@ class RawScrollbar extends StatefulWidget {
          'Scrollbar thumb appearance should only be controlled with thumbVisibility, '
          'isAlwaysShown is deprecated.'
        ),
+       assert(
+         !((thumbVisibility == false || isAlwaysShown == false) && trackVisibility == true),
```

Comment:
The assertion is a bit unfriendly. In this case, don't draw the track and documentation of this behavior. What do you think?
At the same time, the `Scrollbar` widget does not assert this, right?

Reply:
The scrollbar widget does not assert this, but the _MaterialScrollbar and CupertinoScrollbar pass these values to the super class RawScrollbar, so they still get the assertion. 
Here, where the user explicitly says 'show me the track, don't show me the thumb' I think it is reasonable to assert. 

When ScrollbarThemeData.trackVisibility (MaterialStateProperty) is used, and the value changes based on different state combinations, it does not assert and just assumes that if the thumb is not visible then the track should not be.

Reply:
That makes sense, thanks for explaining.